### PR TITLE
Source Freshdesk: enable fine tuning rate limiting configurations #1962

### LINKED ISSUE #1962: Source Freshdesk: enable fine tuning rate limiting configurations
## Tell us about the problem you're trying to solve
I'd like to be able to control how many requests per second the Freshdesk connector uses in order to guarantee that syncs from Airbyte do not block other applications reading from Freshdesk. 

## Acceptance Criteria
1. Add a `requests_per_second` parameter which, when set, throttles requests made per second by airbyte

### PR DESCRIPTION
**Pull request by eugene-kulak**
## What
#1962

## How
*Describe the solution*

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py [hunk 1]
```diff
@@ -107,17 +114,40 @@ def get(self, url: str, params: Mapping = None):
         response = self._session.get(self._api_prefix + url, params=params)
         return self._parse_and_handle_errors(response)
 
+    def consume_credit(self, credit):
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py:117]: This logic is complex. Couldn't we make this a lot simpler by just keeping two numbers: `current_time_window_start` and `current_window_credits_consumed` then running something like: 

```
if not self._requests_per_minute: 
  return 



# Reset time window if it has elapsed
if time.time() > self.current_time_window_start + 60:
  self.current_time_window_start = time.time()
  self.current_window_credits_consumed = 0

if self.current_window_credits_consumed >= self._requests_per_minute
  logger.trace(f"Reached call limit for this minute, wait for {sleep_time:.2f} seconds")
  sleep(max(1, 60 - (time.time() - self.current_window_start_time))
```

or is there an advantage to the current approach

File: airbyte-integrations/connectors/source-freshdesk/source_freshdesk/spec.json [hunk 2]
```diff
@@ -17,6 +17,10 @@
         "type": "string",
         "description": "Freshdesk API Key. See the <a href=\"https://docs.airbyte.io/integrations/sources/freshdesk\">docs</a> for more information on how to obtain this key.",
         "airbyte_secret": true
+      },
+      "requests_per_minute": {
+        "type": "integer",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-freshdesk/source_freshdesk/spec.json:22]:
```suggestion
        "title": "Requests per minute",
        "type": "integer",
```

**Review by sherifnada — APPROVED** [review]
one comment on simplifying the logic

**Comment by sherifnada** [comment]
Once you address the comment, can you bump the version, comment `/publish connector=connectors/source-freshdesk` and merge this PR?

File: airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py (hunk 1 above)
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py:114]: ok, I like your approach, original logic was taken from a python client,
and I was sure that it is fine to handle calls with the same price, but it is not good enough for calls where the price could be 3, 5, etc

**Pull request merged**